ui/component/or-asset-tree/src/index.ts

```diff
@@ -538,6 +568,99 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
         this._onNodeClicked(null, null);
     }
 
+    /**
+     * On Cancel button click, we clear everything and reset the asset tree to original content
+     *
+     * @param event
+     * @protected
+     */
+    protected _onCancelFilterChange() {
+        this._filterValue = '';
+        this._foundAssets = [];
+
+        this._loadAssets();
+    }
+
+    protected _onFilterValueChange(e: OrInputChangedEvent) {
+        this._filterValue = e.detail.value;
+        const oldFilterValue: string = e.detail.previousValue;
+
+        if (!this._filterValue.startsWith(oldFilterValue) || this._filterValue.length === 0) {
+            this._onCancelFilterChange();
+        }
+
+        if (this._searchInputTimer) {
+            clearTimeout(this._searchInputTimer);
+        }
+
+        this._searchInputTimer = setTimeout(async () => {
+            if (this._nodes && this._filterValue.length > 2) {
```

Comment:
With async ops like this it's worth checking `isConnected` status of component on completion to avoid doing work if the component is no longer part of the DOM

Reply:
indeed, I forget that point